src/Text/Pandoc/Readers/Man.hs

```diff
@@ -0,0 +1,284 @@
+      let inls = roffInline roffSt lnpart
+      others <- backSlash <|> return []
+      return $ inls ++ others
+    
+    backSlash :: PandocMonad m => ManParser m [Inline]
+    backSlash = do
+      char '\\'
+      esc <- choice [ char 'f' >> fEscape
+                    , char '-' >> return (Just '-')
+                    , char '%' >> return Nothing
+                    , Just <$> noneOf "\n"
+                    ]
+      ManState {rState = roffSt} <- getState
+      case esc of
+        Just c -> let inls = roffInline roffSt [c]
+                  in parseLineParts >>= (\oth -> return $ inls ++ oth)
+        Nothing -> parseLineParts
+      where
+      
+      fEscape :: PandocMonad m => ManParser m (Maybe Char)
+      fEscape = choice [ char 'B' >> modifyRoffState (\rst -> rst {fontKind = Bold})
+                       , char 'I' >> modifyRoffState (\rst -> rst {fontKind = Italic})
+                       , (char 'P' <|> anyChar) >> modifyRoffState (\rst -> rst {fontKind = Regular})
+                       ]
+                >> return Nothing
+
+finds :: (a -> Bool) -> [a] -> ([a], [a])
+finds predic els = let matched = finds' els
+  in (matched, drop (length matched) els) where
+  finds' [] = []
+  finds' (e:es) | predic e = e : finds' es
+                | otherwise = []
+
+-- | return (matched, notmatched, others)
+findsBoth :: (a -> Bool) -> [a] -> ([a], [a], [a])
+findsBoth predic els =
+  let (matched, els')     = finds predic els
+      (notmatched, els'') = finds (not . predic) els'
+  in (matched, notmatched, els'')
+
+createParas :: [Block] -> [Block]
+createParas bs = inner bs [] where
+  inner :: [Block] -> [Inline] -> [Block]
+  inner [] inls = plainInlinesToPara inls
+  inner (Plain einls : oth) inls = inner oth (inls ++ joinCode einls)
+  inner (block : oth) inls = (plainInlinesToPara inls ++ [block]) ++ inner oth []
+
+  joinCode :: [Inline] -> [Inline]
+  joinCode inls =
+    let (codes, notcodes) = finds isCode inls
+        codeStr (Code _ s) = s
+        codeStr _          = ""
+        joined = Code nullAttr (concat $ codeStr <$> codes)
+    in if null codes
+         then notcodes
+         else joined : notcodes
+
+  plainInlinesToPara :: [Inline] -> [Block]
+  plainInlinesToPara []   = []
+  plainInlinesToPara inls =
+    let (cds, ncds, oth) = findsBoth isCode inls
+        codeToStr (Code _ s) = s
+        codeToStr _ = ""
+        cbs = if null cds
+                then []
+                else [CodeBlock nullAttr (intercalate "\n" $ codeToStr <$> cds)]
+        paras = [Para (intersperse (Str " ") ncds)]
+    in cbs ++ paras ++ plainInlinesToPara oth
+
+  isCode (Code _ _) = True
+  isCode _ = False
+
+parseMan :: PandocMonad m => ManParser m Pandoc
+parseMan = do
+  blocks <- createParas <$> many (parseMacro <|> parseLine)
+  parserst <- pState <$> getState
+  return $ Pandoc (stateMeta parserst) blocks
+
+-- | Read man (troff) from an input string and return a Pandoc document.
```

Comment:
In pandoc we typically put the main function (readMan, here) at the top of the source file.